### DISCUSSION TITLE
How to use CoreWCF in .NET Core 7 running as Windows Service

### DISCUSSION THREAD

**Discussion by timesb** [CoreWCF/CoreWCF#1256] [Q&A]
It's been years since using WCF, classic .NET Framework as Windows Service.
Last few years have been with .NET Core in the cloud space or as app.

So, I was excited to find CoreWCF that Matt Connew created for .NET Core.
With things moving so fast in .NET Core space, and progress in CoreWCF is as when he and the teams are able, I cannot seem to find any working combination of any CoreWCF example/sample/Getting Started with this combo:

a. .NET Core 7
b. running as a Windows Service
c. using CoreWCF to talk to client front-end via CoreWCF
d. ideally using NetTCP binding for secure message transport - but for now just even basic http
The key is running as a Windows Service. 

It seems the .NET Core 6 using <UseStartup> and the separations all have changed to more concise syntax with .NET Core 7 - now all simply in a 'Program.cs' but I cannot get the samples working as a Windows Service.

My confusion also is when googling - there are overlaps that show IHost vs IWebHost and WebApplicationBuilder vs WebHostBuilder, and now it seems in .NETCore 7 to use WebApplication. But then how to inject/use the right statement(s) to make it a Windows Service. The 'AddWindowsService' or 'UseWindowsService' items seem to be either in the IHost or unavailable.

Any help and clarity is appreciated.

At this point, even .NETCore 6 with separate Program and Startup - but as Windows Service - I could use that too.

**Comment by L-E-son** [comment]
Found this question via the internet.

https://github.com/CoreWCF/samples/pull/26
https://learn.microsoft.com/en-us/dotnet/core/extensions/windows-service
